lib/publiccloud/instance.pm

```diff
@@ -346,59 +355,88 @@ sub wait_for_ssh {
     # until it is not over we will recieve "ssh permission denied (pubkey)" error
     # but it is not good reason to die early because after it will be over DMS will return normal
     # user and error will be resolved.
-    $args{ignore_wrong_pubkey} //= 0;
+    #$args{ignore_wrong_pubkey} //= $args{proceed_on_failure};
     $args{username} //= $self->username();
     $args{public_ip} //= $self->public_ip();
+    $args{systemup_check} //= 1;    # check also system up, defult:true; when false,systemup check not executed,only ssh checked.
```

Comment:
```suggestion
    $args{systemup_check} //= 1;    # If true, check if the system state is up, otherwise only the ssh port will be checked for availability. Default: true
```